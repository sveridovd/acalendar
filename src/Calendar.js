import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import memoize from "memoize-one";

import {
	makeDayMatrix,
	makeYearsMatrix,
	MODE_YEAR,
	MODE_MONTH,
	MODE_DAY
} from "./share/util.js";

import {
	Header,
	Choosen,
	ContentDays,
	ContentMonths,
	ContentYears
} from "./subcomponents/calendar.js";

export class Calendar extends React.Component {

	constructor(props) {
		super(props);

		let mode = MODE_DAY;

		this.today = props.today ? props.today : moment();
		this.months = moment.months();
		this.weekdays = moment.weekdays().map(day => day.substring(0, 3));

		this.state = Object.assign({
			mode,
			showedMoment: (props.date ? moment(props.date) : this.today),
			choosenMoment: (props.date ? moment(props.date) : null),

			// MODE_YEAR
			yearincr: 0
		});

		this.daysInMonth = memoize((showedMoment) => {
			return makeDayMatrix(showedMoment);
		});

		this.years = memoize((showedMoment, yearincr) => {
			return makeYearsMatrix(
				moment(showedMoment).year(showedMoment.year() + yearincr)
			);
		});
	}

	componentDidUpdate(prevProps, prevState, snapshot) {
		if (this.props.date && !this.props.date.isSame(prevProps.date, "days")) {
			this.setState({
				showedMoment: this.props.date,
				choosenMoment: this.props.date
			});
		}
	}

	onPrev() {

		switch(this.state.mode) {
			case MODE_YEAR:
				this.setState(prev => ({yearincr: prev.yearincr - 20}));
				break;
			case MODE_MONTH:
				throw new Error("Forbidden in MODE_YEAR");
			case MODE_DAY:
				this.setState({
					showedMoment: moment(this.state.showedMoment).subtract(1, 'months')
				});
				break;
			default:
				throw new Error("Mode not found: " + this.state.mode);
		}
	}

	onNext() {

		switch(this.state.mode) {
			case MODE_YEAR:
				this.setState(prev => ({yearincr: prev.yearincr + 20}));
				break;
			case MODE_MONTH:
				throw new Error("Forbidden in MODE_MONTH");
				break;
			case MODE_DAY:
				this.setState({
					showedMoment: moment(this.state.showedMoment).add(1, 'months')
				});
				break;
			default:
				throw new Error("Mode not found: " + this.state.mode);
		}
	}

	onChoose(type, val) {

		this.setState(prev => {

			const showedMoment = moment(prev.showedMoment);
			const choosenMoment = moment(prev.choosenMoment || prev.showedMoment);
			let mode = prev.mode;

			switch(type) {
				case "date":
					choosenMoment.date(val);
					choosenMoment.month(showedMoment.month());
					choosenMoment.year(showedMoment.year());
					break;
				case "month":
					showedMoment.month(val);
					choosenMoment.month(val);
					mode = MODE_DAY;
					break;
				case "year":
					showedMoment.year(val);
					choosenMoment.year(val);
					mode = MODE_MONTH;
					break;
				default:
					throw new Error("Type not found");
			}

			return {choosenMoment, showedMoment, mode};
		}, () => {
			this.props.onChange
				&& this.props.onChange(moment(this.state.choosenMoment));
		});
	}

	showMode(mode) {
		this.setState({mode});
	}

	render() {

		const daysInMonth = this.daysInMonth(this.state.showedMoment);
		const years = this.years(this.state.showedMoment, this.state.yearincr);

		return (
			<div className="atcalendar">

				<Choosen
					weekdays={this.weekdays}
					months={this.months}
					choosenMoment={this.state.choosenMoment} />

				<div className="atcalendar__head">
					<button className="atcalendar__head__prev-month"
						disabled={this.state.mode === MODE_MONTH}
						onClick={this.onPrev.bind(this)}/>

					<Header
						mode={this.state.mode}
						months={this.months}
						showedMoment={this.state.showedMoment}
						showMode={this.showMode.bind(this)}
						/>

					<button className="atcalendar__head__next-month"
						disabled={this.state.mode === MODE_MONTH}
						onClick={this.onNext.bind(this)}/>
				</div>

				{
					this.state.mode === MODE_DAY &&
						<ContentDays
							weekdays={this.weekdays}
							daysInMonth={daysInMonth}
							showedMoment={this.state.showedMoment}
							choosenMoment={this.state.choosenMoment}
							onChooseDate={this.onChoose.bind(this, "date")}
							/>
				}

				{
					this.state.mode === MODE_MONTH &&
					<ContentMonths
						months={this.months}
						onChooseMonth={this.onChoose.bind(this, "month")}
						/>
				}

				{
					this.state.mode === MODE_YEAR &&
					<ContentYears
						years={years}
						onChooseYear={this.onChoose.bind(this, "year")}
						/>
				}

			</div>
		);
	}

}

Calendar.propTypes = {
	date: PropTypes.object,
	onChange: PropTypes.func
};

