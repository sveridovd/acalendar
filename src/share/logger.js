export function debug(name) {
	return (...rest) => {
		if (process.env.ENV === "development")
			console.log(name, ":", ...rest);
	};
}