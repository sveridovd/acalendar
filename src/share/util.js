import moment from "moment";
import memoize from "memoize-one";

export const MODE_YEAR = "YEAR";
export const MODE_MONTH = "MONTH";
export const MODE_DAY = "DAY";

export const MATRIX_COUNT_ELEMENTS = 42;
export const MATRIX_EMPTY_ELEMENT_VAL = "-";

export function cssClassHelper(rules) {
	let str = "";
	Object.keys(rules).forEach((className) => {
		str += rules[className] ? className + " " : "";
	});
	return str.trim();
}

export const generateWithValue = memoize(function (count, val) {
	let array = [];
	for (let i = 0; i < count; ++i) {
		array.push(val);
	}
	return array;
});

export function generateWithFunction(count, func) {
	let array = [];
	for (let i = 0; i < count; ++i) {
		array.push(func(i));
	}
	return array;
}

export function makeDayMatrix(showedMoment) {
	const startDay = moment(showedMoment).date(1).day();
	const countOfDate = showedMoment.daysInMonth();

	return generateWithValue(startDay, MATRIX_EMPTY_ELEMENT_VAL)
			.concat(generateWithFunction(countOfDate, (i) => i + 1))
			.concat(generateWithValue(MATRIX_COUNT_ELEMENTS - startDay - countOfDate, MATRIX_EMPTY_ELEMENT_VAL));
}

export function makeYearsMatrix(showedMoment) {
	const currentYear = showedMoment.year();
	return Array.from(generateWithFunction(20, (i) => i).reverse(), (x) => currentYear - x);
}

export function formatNumberOfMonth(i) {
	return i < 10 ? `0${i}` : i;
}