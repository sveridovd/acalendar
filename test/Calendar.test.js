import React from "react";
import Renderer from "react-test-renderer";
import moment from "moment";
import { Calendar } from "../src/Calendar.js";
import {MODE_MONTH, MODE_YEAR} from "../src/share/util";

class CalendarTest extends React.Component {

	constructor(props) {
		super(props);

		this.state = {
			date: props.date
		};
	}

	onChange(date) {
		this.setState(date);
	}

	render() {
		return <Calendar onChange={this.onChange.bind(this)} date={this.state.date} />
	}
}

test("Calendar", () => {
	const renderer = Renderer.create(<CalendarTest date={moment("13/10/2019 12:00", "DD/MM/YYYY hh:mm")}/>);
	const root = renderer.root;
	const calendarTest = root.instance;
	const calendar = root.findByType(Calendar).instance;

	{
		let atcalendar__head__center__date = root.findByProps({
			className: "atcalendar__head__center__date"
		});
		expect(atcalendar__head__center__date.children[0].children[0]).toBe("October");
		expect(atcalendar__head__center__date.children[1].children[0]).toBe("2019");
		expect(renderer.toJSON()).toMatchSnapshot();
	}


	{
		calendar.showMode(MODE_MONTH);
		let atcalendar__head__center__month = root.findByProps({
			className: "atcalendar__head__center__month"
		});
		expect(atcalendar__head__center__month.children[0].children[0]).toBe("2019");
		expect(renderer.toJSON()).toMatchSnapshot();
	}


	{
		calendar.showMode(MODE_YEAR);
		let atcalendar__head__center__year = root.findByProps({
			className: "atcalendar__head__center__year"
		});
		expect(atcalendar__head__center__year.children[0].children[0]).toBe("Choose year");
		expect(renderer.toJSON()).toMatchSnapshot();
	}


	{
		calendar.onChoose("year", 2017);
		let atcalendar__head__center__month = root.findByProps({
			className: "atcalendar__head__center__month"
		});
		expect(atcalendar__head__center__month.children[0].children[0]).toBe("2017");
		expect(renderer.toJSON()).toMatchSnapshot();
	}

	{
		calendar.onChoose("month", "December");
		let atcalendar__head__center__date = root.findByProps({
			className: "atcalendar__head__center__date"
		});
		expect(atcalendar__head__center__date.children[0].children[0]).toBe("December");
		expect(atcalendar__head__center__date.children[1].children[0]).toBe("2017");
		expect(renderer.toJSON()).toMatchSnapshot();
	}
});
