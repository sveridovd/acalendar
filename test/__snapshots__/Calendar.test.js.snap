// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar 1`] = `
<div
  className="atcalendar"
>
  <div
    className="atcalendar__choosen"
  >
    <div
      className="atcalendar__choosen__icon"
    />
    <div
      className="atcalendar__choosen__date"
    >
      <span>
        October
         
        2019
        , 
        Sun
         
        13
      </span>
    </div>
  </div>
  <div
    className="atcalendar__head"
  >
    <button
      className="atcalendar__head__prev-month"
      disabled={false}
      onClick={[Function]}
    />
    <div
      className="atcalendar__head__center"
    >
      <div
        className="atcalendar__head__center__date"
      >
        <span
          className="clickable"
          onClick={[Function]}
        >
          October
        </span>
        <span
          className="clickable"
          onClick={[Function]}
        >
          2019
        </span>
      </div>
    </div>
    <button
      className="atcalendar__head__next-month"
      disabled={false}
      onClick={[Function]}
    />
  </div>
  <ul
    className="atcalendar__week-days"
  >
    <li
      className="atcalendar__week-days__item"
    >
      Sun
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Mon
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Tue
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Wed
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Thu
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Fri
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Sat
    </li>
  </ul>
  <ul
    className="atcalendar__dates"
  >
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      3
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      4
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      5
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      6
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      7
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      8
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      9
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      10
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      11
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      12
    </li>
    <li
      className="atcalendar__dates__item choosen"
      onClick={[Function]}
    >
      13
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      14
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      15
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      16
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      17
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      18
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      19
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      20
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      21
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      22
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      23
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      24
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      25
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      26
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      27
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      28
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      29
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      30
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      31
    </li>
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
  </ul>
</div>
`;

exports[`Calendar 2`] = `
<div
  className="atcalendar"
>
  <div
    className="atcalendar__choosen"
  >
    <div
      className="atcalendar__choosen__icon"
    />
    <div
      className="atcalendar__choosen__date"
    >
      <span>
        October
         
        2019
        , 
        Sun
         
        13
      </span>
    </div>
  </div>
  <div
    className="atcalendar__head"
  >
    <button
      className="atcalendar__head__prev-month"
      disabled={true}
      onClick={[Function]}
    />
    <div
      className="atcalendar__head__center"
    >
      <div
        className="atcalendar__head__center__month"
      >
        <span
          onClick={[Function]}
        >
          2019
        </span>
      </div>
    </div>
    <button
      className="atcalendar__head__next-month"
      disabled={true}
      onClick={[Function]}
    />
  </div>
  <ul
    className="atcalendar__months"
  >
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      January
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        01
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      February
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        02
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      March
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        03
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      April
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        04
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      May
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        05
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      June
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        06
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      July
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        07
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      August
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        08
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      September
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        09
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      October
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        10
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      November
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        11
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      December
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        12
      </sup>
    </li>
  </ul>
</div>
`;

exports[`Calendar 3`] = `
<div
  className="atcalendar"
>
  <div
    className="atcalendar__choosen"
  >
    <div
      className="atcalendar__choosen__icon"
    />
    <div
      className="atcalendar__choosen__date"
    >
      <span>
        October
         
        2019
        , 
        Sun
         
        13
      </span>
    </div>
  </div>
  <div
    className="atcalendar__head"
  >
    <button
      className="atcalendar__head__prev-month"
      disabled={false}
      onClick={[Function]}
    />
    <div
      className="atcalendar__head__center"
    >
      <div
        className="atcalendar__head__center__year"
      >
        <span>
          Choose year
        </span>
      </div>
    </div>
    <button
      className="atcalendar__head__next-month"
      disabled={false}
      onClick={[Function]}
    />
  </div>
  <ul
    className="atcalendar__years"
  >
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2000
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2001
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2002
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2003
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2004
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2005
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2006
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2007
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2008
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2009
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2010
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2011
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2012
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2013
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2014
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2015
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2016
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2017
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2018
    </li>
    <li
      className="atcalendar__years__item"
      onClick={[Function]}
    >
      2019
    </li>
  </ul>
</div>
`;

exports[`Calendar 4`] = `
<div
  className="atcalendar"
>
  <div
    className="atcalendar__choosen"
  >
    <div
      className="atcalendar__choosen__icon"
    />
    <div
      className="atcalendar__choosen__date"
    >
      <span>
        October
         
        2017
        , 
        Fri
         
        13
      </span>
    </div>
  </div>
  <div
    className="atcalendar__head"
  >
    <button
      className="atcalendar__head__prev-month"
      disabled={true}
      onClick={[Function]}
    />
    <div
      className="atcalendar__head__center"
    >
      <div
        className="atcalendar__head__center__month"
      >
        <span
          onClick={[Function]}
        >
          2017
        </span>
      </div>
    </div>
    <button
      className="atcalendar__head__next-month"
      disabled={true}
      onClick={[Function]}
    />
  </div>
  <ul
    className="atcalendar__months"
  >
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      January
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        01
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      February
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        02
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      March
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        03
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      April
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        04
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      May
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        05
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      June
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        06
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      July
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        07
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      August
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        08
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      September
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        09
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      October
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        10
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      November
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        11
      </sup>
    </li>
    <li
      className="atcalendar__months__item"
      onClick={[Function]}
    >
      December
      <sup
        className="atcalendar__months__item__number-of-month"
      >
         
        12
      </sup>
    </li>
  </ul>
</div>
`;

exports[`Calendar 5`] = `
<div
  className="atcalendar"
>
  <div
    className="atcalendar__choosen"
  >
    <div
      className="atcalendar__choosen__icon"
    />
    <div
      className="atcalendar__choosen__date"
    >
      <span>
        December
         
        2017
        , 
        Wed
         
        13
      </span>
    </div>
  </div>
  <div
    className="atcalendar__head"
  >
    <button
      className="atcalendar__head__prev-month"
      disabled={false}
      onClick={[Function]}
    />
    <div
      className="atcalendar__head__center"
    >
      <div
        className="atcalendar__head__center__date"
      >
        <span
          className="clickable"
          onClick={[Function]}
        >
          December
        </span>
        <span
          className="clickable"
          onClick={[Function]}
        >
          2017
        </span>
      </div>
    </div>
    <button
      className="atcalendar__head__next-month"
      disabled={false}
      onClick={[Function]}
    />
  </div>
  <ul
    className="atcalendar__week-days"
  >
    <li
      className="atcalendar__week-days__item"
    >
      Sun
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Mon
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Tue
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Wed
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Thu
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Fri
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Sat
    </li>
  </ul>
  <ul
    className="atcalendar__dates"
  >
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      3
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      4
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      5
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      6
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      7
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      8
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      9
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      10
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      11
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      12
    </li>
    <li
      className="atcalendar__dates__item choosen"
      onClick={[Function]}
    >
      13
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      14
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      15
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      16
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      17
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      18
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      19
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      20
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      21
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      22
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      23
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      24
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      25
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      26
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      27
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      28
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      29
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      30
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      31
    </li>
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
  </ul>
</div>
`;

exports[`Calendar 6`] = `
<div
  className="atcalendar"
>
  <div
    className="atcalendar__choosen"
  >
    <div
      className="atcalendar__choosen__icon"
    />
    <div
      className="atcalendar__choosen__date"
    >
      <span>
        November
         
        2013
        , 
        Mon
         
        25
      </span>
    </div>
  </div>
  <div
    className="atcalendar__head"
  >
    <button
      className="atcalendar__head__prev-month"
      disabled={false}
      onClick={[Function]}
    />
    <div
      className="atcalendar__head__center"
    >
      <div
        className="atcalendar__head__center__date"
      >
        <span
          className="clickable"
          onClick={[Function]}
        >
          November
        </span>
        <span
          className="clickable"
          onClick={[Function]}
        >
          2013
        </span>
      </div>
    </div>
    <button
      className="atcalendar__head__next-month"
      disabled={false}
      onClick={[Function]}
    />
  </div>
  <ul
    className="atcalendar__week-days"
  >
    <li
      className="atcalendar__week-days__item"
    >
      Sun
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Mon
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Tue
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Wed
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Thu
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Fri
    </li>
    <li
      className="atcalendar__week-days__item"
    >
      Sat
    </li>
  </ul>
  <ul
    className="atcalendar__dates"
  >
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      3
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      4
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      5
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      6
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      7
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      8
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      9
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      10
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      11
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      12
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      13
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      14
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      15
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      16
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      17
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      18
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      19
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      20
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      21
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      22
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      23
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      24
    </li>
    <li
      className="atcalendar__dates__item choosen"
      onClick={[Function]}
    >
      25
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      26
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      27
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      28
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      29
    </li>
    <li
      className="atcalendar__dates__item"
      onClick={[Function]}
    >
      30
    </li>
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
    <li
      className="atcalendar__dates__item empty"
    />
  </ul>
</div>
`;
